assets/src/infrastructure/ui/display/AlertDialog.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {
+	AlertDialog as ChakraAlertDialog,
+	AlertDialogBody,
+	AlertDialogFooter,
+	AlertDialogHeader,
+	AlertDialogContent,
+	AlertDialogOverlay,
+	Button,
+} from '@chakra-ui/core';
+import { __ } from '@wordpress/i18n';
+
+import { EspressoButton } from '@application/ui/input';
+import { AlertDialogProps } from './types';
+
+const AlertDialog: React.FC<AlertDialogProps> = ({ buttonProps, onConfirm, title }) => {
+	const [isOpen, setIsOpen] = React.useState(false);
+	const cancelRef = React.useRef();
+	const onClose = () => setIsOpen(false);
+	const onDelete = () => {
+		onClose();
+		onConfirm();
+	};
+
+	return (
+		<>
+			<EspressoButton {...buttonProps} onClick={() => setIsOpen(true)} />
+
+			<ChakraAlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
+				<AlertDialogOverlay />
+				<AlertDialogContent>
+					{buttonProps?.tooltip && (
+						<AlertDialogHeader fontSize='lg' fontWeight='bold' textTransform='capitalize'>
+							{buttonProps.tooltip}
+						</AlertDialogHeader>
+					)}
+					<AlertDialogBody>{title}</AlertDialogBody>
+					<AlertDialogFooter>
+						<Button ref={cancelRef} onClick={onClose}>
+							{__('No')}
+						</Button>
+						<Button variantColor='red' onClick={onDelete} ml={3}>
+							{__('Yes')}
+						</Button>
+					</AlertDialogFooter>
+				</AlertDialogContent>
+			</ChakraAlertDialog>
+		</>
+	);
+};
+
+export default AlertDialog;
```

Comment:
`AlertDialog` should also handle the normal error alerts which may not necessarily have footer buttons or may be just the `OK` button. Also a prop like `alertType` could be passed as `success|error|warning|info` and an appropriate icon could be used in `AlertDialogHeader`

Reply:
An example of the above is the TAM alert dialog when there are orphan entities.

Reply:
I understand this, but based on `AlertDialog` docs:
```
AlertDialog and it's components composes the Modal component, the only exception is that it requires a leastDestructiveRef which is similar to the initialFocusRef in Modal
```
I think it's better to leave this component as it is to avoid pushing too much logic here and instead create another component for the use case you described, but which would use `Modal` instead.
What do you think?

Reply:
Yeah I am OK with that.